includes/classes/class-fixinsecurecontent.php

```diff
@@ -0,0 +1,343 @@
+<?php
+/**
+ * Handles the fixing of insecure content in posts for REST and WP CLI requests.
+ *
+ * @package ICW
+ */
+
+namespace ICW\FIX;
+
+use WP_CLI;
+use WP_CLI\Utils;
+use WP_CLI_Command;
+use WP_Query;
+
+/**
+ * FixInsecureContent class.
+ */
+class FixInsecureContent {
+	/**
+	 * Is current run a dry run?
+	 *
+	 * @var bool
+	 */
+	private bool $dry_run;
+
+	/**
+	 * Total number of posts scanned for URLs.
+	 *
+	 * @var int
+	 */
+	private int $total_post_count = 1;
+
+	/**
+	 * Data of posts where 1 or more URLs were found to be insecure and command fixed summary.
+	 *
+	 * @var array
+	 */
+	private array $fixed_post_count = array();
+
+	/**
+	 * Array of warning messages.
+	 *
+	 * @var array
+	 */
+	private array $warning_count = array();
+
+	/**
+	 * Class instance
+	 *
+	 * @var ?FixInsecureContent
+	 */
+	protected static ?FixInsecureContent $instance = null;
+
+	/**
+	 * Get class instance
+	 *
+	 * @return FixInsecureContent
+	 */
+	public static function get_instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Gets the "post_type" args for the WP_Query calls for the functionality to fix insecure content from within the admin UI.
+	 *
+	 * @return array
+	 */
+	public function get_wp_query_post_type_args(): array {
+		return array(
+			'show_ui'      => true,
+			'public'       => true,
```

Comment:
```suggestion
```

I think the other two will be fine, the user may have permission to edit some private post types

Reply:
Done.